components/webui/imports/ui/SearchView/datetime.js

```diff
@@ -80,45 +65,149 @@ const computeTimeRange = (token) => () => {
             TIME_RANGE_MODIFIER.TO_DATE,
         ].includes(modifier);
 
-        endTime = (true === isEndingNow) ?
-            DateTime.utc() :
-            DateTime.utc().minus({[unit]: amount}).endOf(unit);
-        beginTime = (true === isBeginStartOfUnit) ?
-            endTime.startOf(unit) :
-            endTime.minus({[unit]: amount});
+        end = (true === isEndingNow) ?
+            dayjs.utc() :
+            dayjs.utc().subtract(amount, unit)
+                .endOf(unit);
+        begin = (true === isBeginStartOfUnit) ?
+            end.startOf(unit) :
+            end.subtract(amount, unit);
     }
 
     return {
-        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
-        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+        begin,
+        end,
     };
 };
 
+const DEFAULT_TIME_RANGE = computeTimeRange(
+    `${TIME_UNIT.ALL}_${TIME_RANGE_MODIFIER.NONE}_0`,
+);
+
+// TODO: Switch date pickers so we don't have to do this hack
 /**
- * Changes the timezone of a given Date object to UTC without changing the time.
+ * Converts a Dayjs object in UTC, to a JavaScript Date object that represents the same date and
+ * time, but aligned with the local timezone.
+ * This is achieved without altering the original timestamp values (year, month, day, hour,
+ * minute, second, and millisecond).
  *
- * @param {Date} date Date object to convert to UTC
- * @returns {Date} A new Date object with the same time values in UTC timezone
+ * @param {dayjs.Dayjs} utcDatetime
+ * @return {Date | undefined} The corresponding Date object, or `undefined` when `utcDatetime`
+ * is `undefined`.
  */
-const changeTimezoneToUtcWithoutChangingTime = (date) => {
-    return new Date(Date.UTC(
-        date.getFullYear(),
-        date.getMonth(),
-        date.getDate(),
-        date.getHours(),
-        date.getMinutes(),
-        date.getSeconds(),
-        date.getMilliseconds(),
-    ));
+const convertUtcToSameLocalDate = (utcDatetime) => {
+    const localTz = dayjs.tz.guess();
+    return utcDatetime?.tz(localTz, true).toDate();
 };
 
-const DEFAULT_TIME_RANGE = computeTimeRange(
-    `${TIME_RANGE_UNIT.ALL}_${TIME_RANGE_MODIFIER.NONE}_0`,
-);
+/**
+ * Converts a JavaScript Date object in local timezone, to a Dayjs object that represents the same
+ * date and time, but aligned with the UTC.
+ *
+ * @param {Date} localDate
+ * @return {dayjs.Dayjs} The corresponding Dayjs object
+ */
+const convertLocalToSameUtcDatetime = (localDate) => {
+    return dayjs(localDate)?.utc(true);
+};
+
+/**
+ * Converts a Unix milliseconds timestamp of a JavaScript Date object in local timezone, to a Dayjs
+ * object that represents the same date and time, but aligned with the UTC.
+ *
+ * @param {number} localTimeStampUnixMs
+ * @return {dayjs.Dayjs} The corresponding Dayjs object
+ */
+const convertLocalUnixMsToSameUtcDatetime = (localTimeStampUnixMs) => {
```

Comment:
> Just to confirm, Datetime is how you're referring to the Dayjs object?

Yes. 

> It's a bit weird to say LocalUnix since Unix timestamps should be with respect to the Unix epoch which is relative to UTC already.

Indeed, I struggled a bit to name this function and the parameters, given the use case is quite tricky. (and still I don't think I have properly named it)

Note that the Chart.js timescale operates in local time zone, and we do have to convert the utc timestamps into the local time in the chart's `parser`. However, the `zoom` plugin does not take the `parser` effect into account and reverse the conversion before updating the `min, max` of the xAxis. That is, if I zoom into a time range starting from 00:00 January 1, 1970 / Toronto, the `min` (`begin`) of the range would have a value of `36000000` (`1970-01-01T05:00:00-05:00`), and we need to map it to `0` which is the unix epoch, i.e. 00:00 January 1, 1970 / UTC.

Two iterations of reversion are needed. Here is the prototyping code:
```js
const dayjs = require("dayjs");

dayjs.extend(require("dayjs/plugin/utc"));
dayjs.extend(require("dayjs/plugin/timezone"));

const original = dayjs(36000000);
console.log("original:", original.valueOf(), original.format(), original);

const localTz = dayjs.tz.guess();
console.log(localTz);

// Iter 1: mark it as local date same timestamp value but add '$timezone': 'America/Toronto'
const localDatetime = original.tz(localTz);
console.log("localDatetime:", localDatetime.valueOf(), localDatetime.format(), localDatetime);

// Iter 1: convert to utc while keeping the local time
const utcDatetime = localDatetime.utc(true);
console.log("utcDatetime:", utcDatetime.valueOf(), utcDatetime.format(), utcDatetime);

// Iter 2: mark it as local date same timestamp value but add '$timezone': 'America/Toronto'
const localDatetime2 = utcDatetime.tz(localTz);
console.log("localDatetime2:", localDatetime2.valueOf(), localDatetime2.format(), localDatetime2);

// Iter 2: convert to utc while keeping the local time
const utcDatetime2 = localDatetime2.utc(true);
console.log("utcDatetime2:", utcDatetime2.valueOf(), utcDatetime2.format(), utcDatetime2);
```

Feel free to propose a better name / an alternative handling.